### PR TITLE
Fuse lm_head with cross entropy loss to reduce memory usage and speed up

### PR DESCRIPTION
**Pull request by Hz188**
# Description
对lm head和交叉熵损失计算过程进行算子融合，避免物化全量logits和中间变量，节约显存，同时算子级别加速

### CODE REVIEW

**Review by From00 — APPROVED** [review]
LGTM

**Review by Waynezee — APPROVED** [review]
